### PR TITLE
Strict type `Dependabot::Hex::FileFetcher`

### PR DESCRIPTION
**Pull request by raj-meka**
### What are you trying to accomplish?
Sorbet Implementation for Hex FileFetcher ruby file.

### Anything you want to highlight for special attention from reviewers?
Please pay attention to the fixes in the Hex FileFetcher ruby file.

### How will you know you've accomplished your goal?
The sorbet rules will fire as needed.

### Checklist
 

- [x] I have run the complete test suite to ensure all tests and linters pass.
- [x]  I have thoroughly tested my code changes to ensure they work as expected, including adding additional tests for new functionality.
- [x]  I have written clear and descriptive commit messages.
- [x]  I have provided a detailed description of the changes in the pull request, including the problem it addresses, how it fixes the problem, and any relevant details about the implementation.
- [x]  I have ensured that the code is well-documented and easy to understand.

### CODE REVIEW

File: hex/lib/dependabot/hex/file_fetcher.rb [hunk 1]
```diff
@@ -37,36 +39,40 @@ def fetch_files
 
       private
 
+      sig { returns(T.nilable(DependencyFile)) }
       def mixfile
-        @mixfile ||= fetch_file_from_host("mix.exs")
+        fetch_file_from_host("mix.exs")
```
Comment (JamieMagee) [inline hex/lib/dependabot/hex/file_fetcher.rb:44]: Why did `@mixfile` get removed?

File: hex/lib/dependabot/hex/file_fetcher.rb [hunk 2]
```diff
@@ -86,18 +92,20 @@ def subapp_mixfiles
...
-            path = Pathname.new(File.join(*support_file_args.compact.reverse))
-                           .cleanpath.to_path
+          mixfile&.content&.gsub("__DIR__", "\"#{mixfile_dir}\"")&.scan(SUPPORT_FILE)&.map do |support_file_args|
+            path = Pathname.new(File.join(Array(support_file_args).compact.reverse))
+                           .cleanpath
+                           .to_path
             fetch_file_from_host(path).tap { |f| f.support_file = true }
           end
-        end
+        end.compact
```
Comment (JamieMagee) [inline hex/lib/dependabot/hex/file_fetcher.rb:108]: Why is this required?

**Review by GarryHurleyJr — APPROVED** [review]
adding type and signature checking to method definitions will make for better identification of possible code defects in the future.

**Review by JamieMagee — APPROVED** [review]

File: hex/lib/dependabot/hex/file_fetcher.rb [hunk 3]
```diff
@@ -37,36 +39,41 @@ def fetch_files
 
       private
 
+      sig { returns(T.nilable(DependencyFile)) }
       def mixfile
-        @mixfile ||= fetch_file_from_host("mix.exs")
+        @mixfile ||= T.let(fetch_file_from_host("mix.exs"), T.nilable(Dependabot::DependencyFile))
+        fetch_file_from_host("mix.exs")
```
Comment (landongrindheim) [inline hex/lib/dependabot/hex/file_fetcher.rb:45]: 🔎 I assume memoization is being used here because `fetch_file_from_host` is expensive. Line 44 does this memoization, but the same method is called again on line  45 (negating memoization)
